service/src/main/java/bio/terra/pipelines/app/controller/JobsApiController.java

```diff
@@ -0,0 +1,131 @@
+
+  private SamUser getAuthenticatedInfo() {
+    return samUserFactory.from(request, samConfiguration.basePath());
+  }
+
+  // -- Jobs --
+
+  @Override
+  public ResponseEntity<ApiCreatedJob> createJob(
+      @PathVariable("pipelineId") String pipelineId, @RequestBody ApiCreateJobRequestBody body) {
+    final SamUser userRequest = getAuthenticatedInfo();
+    String userId = userRequest.getSubjectId();
+    String pipelineVersion = body.getPipelineVersion();
+
+    logger.info(
+        "Creating {} pipeline job (version {}) for {} subject {}",
+        pipelineId,
+        pipelineVersion,
+        userRequest.getEmail(),
+        userId);
+
+    // TODO revisit the following (this is copy-pasta from WSM)
+    // Unlike other operations, there's no Sam permission required to create a workspace. As long as
+    // a user is enabled, they can call this endpoint.
+
+    JobRequest jobRequest = new JobRequest(pipelineId, pipelineVersion);
+    UUID createdJobUuid = jobsService.createJob(userId, jobRequest);
+
+    ApiCreatedJob createdJobResponse = new ApiCreatedJob();
+    createdJobResponse.setJobId(createdJobUuid.toString());
+    logger.info("Created {} job {}", pipelineId, createdJobUuid);
+
+    return new ResponseEntity<>(createdJobResponse, HttpStatus.OK);
+  }
+
+  @Override
+  public ResponseEntity<ApiJobResponse> getJob(
+      @PathVariable("pipelineId") String pipelineId, @PathVariable("jobId") String jobId) {
+    final SamUser userRequest = getAuthenticatedInfo();
+    String userId = userRequest.getSubjectId();
+    Job job = jobsService.getJob(userId, pipelineId, jobId);
+    ApiJobResponse result = jobToApi(job);
+
+    return new ResponseEntity<>(result, HttpStatus.OK);
+  }
+
+  @Override
+  public ResponseEntity<ApiJobsGetResponse> getJobs(@PathVariable("pipelineId") String pipelineId) {
+    final SamUser userRequest = getAuthenticatedInfo();
+    String userId = userRequest.getSubjectId();
+    List<Job> jobList = jobsService.getJobs(userId, pipelineId);
+    ApiJobsGetResponse result = jobsToApi(jobList);
+
+    return new ResponseEntity<>(result, HttpStatus.OK);
+  }
+
+  static ApiJobResponse jobToApi(Job job) {
+    ApiJobResponse apiResult =
+        new ApiJobResponse()
+            .jobId(job.getJobId().toString())
+            .userId(job.getUserId())
+            .pipelineId(job.getPipelineId())
+            .pipelineVersion(job.getPipelineVersion())
+            .timeSubmitted(job.getTimeSubmitted().toString())
+            .timeCompleted(Objects.requireNonNullElse(job.getTimeCompleted(), "").toString())
+            .status(job.getStatus());
+
+    return apiResult;
+  }
+
+  static ApiJobsGetResponse jobsToApi(List<Job> jobList) {
+    ApiJobsGetResponse apiResult = new ApiJobsGetResponse();
+
+    for (Job job : jobList) {
+      var apiJob = jobToApi(job);
+
+      apiResult.add(apiJob); // is there a better function to use here? e.g. addJobItem()
```

Comment:
TODO: look at this - find the example from template

Reply:
I couldn't actually find this in either the template project or TPS. so I'll leave it as is (and remove the comment)